### DISCUSSION TITLE
Declare Enum when declaring a function, as its returnType

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5616] [Language Ideas]
It's convenient when functions are returning status instead of data.
Instead of declaring enum then using it as a return type this is cleaner

**Comment by canton7** [comment]
Can you give an example?

**Comment by Richiban** [comment]
This sounds like you'd like an anonymous, inline enum if I have understood you correctly?

So you'd be able to write:

```cs
Public (enum { Ok, Error }) GetStatus()
{ 
...
}
```

There is at least partly related to Type Unions at https://github.com/dotnet/csharplang/discussions/399

**Reply by Xyncgas** [reply]
yeah

**Reply by DavidArno** [reply]
Indeed, if ad hoc unions end up being supported, then we might suppose that:
```c#
public Ok | Error  GetStatus()
{
    ...
}
```
would be shorthand for something like:
```c#
public enum struct <unspeakable union name>
{
    Ok,
    Error
}
 
public <unspeakable union name> GetStatus()
{
    ...
}
```
and it could then be pattern matched:
```c#
_ = GetStatus() switch {
    Ok => ...
    Error => ...
};
```
